modules/ROOT/pages/custom-menu-toolbar-button.adoc

```diff
@@ -39,4 +40,25 @@ This example configures a toolbar menu button with the label `+My Button+` that
 
 The `+fetch+` function is called when the toolbar menu button's menu is opened. It is a function that takes a callback and passes it an array of menu items to be rendered in the drop-down menu. This allows for asynchronous fetching of the menu items.
 
+[[searchable-menu-buttons]]
+== Searchable Menu Buttons
+
+include::partial$misc/admon-requires-6.2v.adoc[]
+
+The button's menu can be configured to have an input field for searching, as well as its usual items. The presence of the input field is controlled by the `+search+` field in the xref:options[options]. The `+search+` field can be a `+boolean+` or an `+object+` containing a single optional string `placeholder`. By default, `+search+` is false. If `+search+` is not false, the button's menu will contain an input field with any specified placeholder text. As the user types into this field, the `+fetch+` function will be called with the text in the input field passed back as part of `+fetchContext+`. The `+fetch+` function is responsible for using this `+fetchContext+` to determine which items to pass to its `+success+` callback.
+
+The `+fetchContext+` is an object containing a single string property: `+pattern+`. If the toolbar menu button has not configured `search` to be active, then the `+pattern+` string will be empty.
+
+== Searchable menu button example and explanation
```

Comment:
```suggestion
=== Searchable menu button example and explanation
```
This currently has the same heading level as `Searchable Menu Buttons`

Reply:
Yeah, that was intentional, because the other demo has this level. I think we probably just need to have a team discussion about how we want this file setup.